automation/orchestrator.go

```diff
@@ -295,28 +318,31 @@ func (o *Orchestrator) CreateWorkflow(did string, pid profile.ID, triggerOpts []
 	return wf, nil
 }
 
+// SaveWorkflow updates a workflow in the workflow.Store
+func (o *Orchestrator) SaveWorkflow(wf *workflow.Workflow) (*workflow.Workflow, error) {
+	return o.workflows.Put(wf)
+}
```

Comment:
Thinking I should move the checks for empty fields and mismatched fields into this function to allow the workflow store methods to be even simpler, just dumbly putting the entry into the store.